### PR TITLE
feat: support react 18

### PR DESCRIPTION
**Pull request by mihkeleidast**
Adds support for React 18. May seem like a lot of changes, but most of it is actually:
- refactoring tests from enzyme to testing-library
- migrating from tslint to eslint and updating prettier, so a lot of automatic reformatting

The real "incompatibility" with React 18 was due to missing `children` prop on some components.

### CODE REVIEW

File: example/src/demos/htmlCluster.tsx [hunk 1]
```diff
@@ -3,7 +3,10 @@ import ReactMapboxGl, { Marker, Cluster, Popup } from '../../../';
 import styled from 'styled-components';
 
 // tslint:disable-next-line:no-var-requires
```
Comment (SiiMeR) [inline example/src/demos/htmlCluster.tsx]: A small thing to do but can you also remove all `tslint` comments?

File: package.json [hunk 2]
```diff
@@ -75,35 +72,35 @@
   "peerDependencies": {
     "mapbox-gl": "^1.12.0",
     "prop-types": "^15.6.2",
-    "react": "^16.11.0 || ^17.0.2",
-    "react-dom": "^16.11.0 || ^17.0.2"
+    "react": "^16.11.0 || ^17.0.2 || ^18.2.0",
```
Comment (SiiMeR) [inline package.json:75]: Should we remove 16 from this union as I think all of our consumers are on React 17?

Comment (mihkeleidast) [reply package.json:75]: Removing the lower end of a peerDep range would be a breaking change.

File: example/src/demos/htmlCluster.tsx (hunk 1 above)
Comment (mihkeleidast) [reply example/src/demos/htmlCluster.tsx]: Yeah, sure, I actually thought it removed them automatically but seems only some disappeared.

**Review by SiiMeR — APPROVED** [review]
👍

**Pull request merged**